mlir/lib/Dialect/Rock/utility/AccelEmitter.cpp

```diff
@@ -1194,96 +1036,24 @@ RegsAsMatrixSubTiles WmmaEmitter::createAccelGemmOperandTransforms(
   }
   // compute block sub tile transforms
   {
-    SmallVector<Attribute> transformAttrs;
-    // First coordinate transform
-    TopDownTMBuilder splitIter(b, {"tid", "iter"},
-                               {blockSize, dRepeats * kpackPerThread * kPack},
-                               loc);
-    {
-      splitIter.passThrough("tid");
-      if (isKContigousDim) {
-        splitIter.merge({"drepeat", "kpack_iter", "kpack"}, {1, 2, 3}, "iter",
-                        {dRepeats, kpackPerThread, kPack});
-      } else {
-        splitIter.merge({"kpack_iter", "drepeat", "kpack"}, {1, 2, 3}, "iter",
-                        {kpackPerThread, dRepeats, kPack});
-      }
-    }
-    TransformMapAttr splitIterAttr = splitIter.get();
-    transformAttrs.push_back(splitIterAttr);
-    // Second coordinate transform
-    TopDownTMBuilder splitTid =
-        TopDownTMBuilder::below(splitIter, splitIterAttr);
-    {
-      splitTid.passThrough({"kpack"}, {0}, {"kpack"});
-      splitTid.merge({"wave_id", "lane_id"}, {1, 2}, "tid",
-                     {blockSize / waveSize, waveSize});
-      splitTid.passThrough({"d_iter", "k_iter"}, {3, 4},
-                           {"drepeat", "kpack_iter"});
-    }
-    TransformMapAttr splitTidAttr = splitTid.get();
-    transformAttrs.push_back(splitTidAttr);
-    // Second coordinate transform
-    TopDownTMBuilder splitWaveId =
-        TopDownTMBuilder::below(splitTid, splitTidAttr);
-    {
-      splitWaveId.passThrough({"kpack"}, {0}, {"kpack"});
-      splitWaveId.merge({"wave_m", "wave_n"}, {1, 2}, "wave_id",
-                        {mWaves, nWaves});
-      splitWaveId.passThrough({"lane_id", "d_iter", "k_iter"}, {3, 4, 5},
-                              {"lane_id", "d_iter", "k_iter"});
+    SetVector<StringRef> dimensionsToRemove;
+    dimensionsToRemove.insert("k_loop");
```

Comment:
Is there no `append` or other method that takes an initializer list?

Reply:
I didn't find any. We could change removeUpperDims to take llvm::StringSet instead of SetVector<StringRef>, it provides a constructor that takes an initializer list and seems the right fit because the inputs are a set of strings.

Reply:
Yeah, we might want to go to `StringSet`.

... which is reminding me that we might want to move the transform builders to `StringSaver`s